contributing/COMMITS.md

```diff
@@ -0,0 +1,242 @@
+# Commit guidelines
+
+## What makes a good commit
+
+A good commit should be a small, self-contained change to a vertical
+slice of the code base, that leaves it in a working state.
+
+For bigger changes, layering functionality from the inside towards a
+public interface may facilitate more granular work. For example:
+
+ 1. A commit to add an (tested) internal method for something.
+ 2. A commit to add (tested) public functionality that uses the new
+    API.
+
+This may seem be at odds with some TDD based workflows. See the
+[TDD workflows and commit granularity](#tdd-workflows-and-commit-granularity)
+section below for more details on this.
+
+
+### Pre-push check list
+
+Before pushing any commits upstream, give them a quick proof read and
+double check:
+
+ - You have a suitable [CLA](PROCESS.md#contribution-sign-off) in
+   place.
+ - All your commits signed-off and signed (using `-s` to accept the
+   [DCO](PROCESS.md#contribution-sign-off), and
+   [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)
+   signed to verify your identity).
+ - Each commit only affect a single logical unit of work.
+ - The documentation been updated accordingly to cover your changes.
+ - There is no dead, commented out or placeholder code.
+ - There are no stale comments from the previous implementation.
+ - There are no stray changes that should be in another commit.
+ - There are no inadvertent white space or other changes to unrelated
+   areas of the files.
+
+If you want to take the opportunity to clean up minor cosmetic issues
+(such as spelling/white space/etc.) as part of your work, this is fine,
+just do them in their own commit. This makes it clear what is an
+intended change vs an accidental one.
+
+
+## Commit Messages
+
+Commit messages are a vital part of code archaeology, they serve as a
+TL;DR and to record rationale for a change. Think of them as
+light-weight decision records.
+
+We use the [50/72](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+format for Commit messages. This aims to make them readable in the
+widest variety of scenarios. In addition, we have a few other
+conventions.
+
+Titles should be no more than 50 characters, and written in the
+imperative tense. If applicable, the component being changed should be
+used to prefix the message (lowercase):
```

Comment:
[fix] "lowercase" required for component prefixes, but examples below are not lowercase.
```suggestion
used to prefix the message:
```